controllers/etcdcluster_controller.go

```diff
@@ -1391,39 +1422,70 @@ func expectedURLForPeer(cluster *etcdv1alpha1.EtcdCluster, peerName string) stri
 	)
 }
 
-func (r *EtcdClusterReconciler) defragCronJob(log logr.Logger, cluster *etcdv1alpha1.EtcdCluster, members *[]etcd.Member) {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*25)
+func (r *EtcdClusterReconciler) defragWithThresholdCronJob(log logr.Logger, cluster *etcdv1alpha1.EtcdCluster) {
+	ctx, cancel, etcdClient, members, ok := r.setupDefragDeps(log, cluster)
+	if !ok {
+		return
+	}
+	defer cancel()
+	defer etcdClient.Close()
+
+	err := defragger.DefragIfNecessary(ctx, r.DefragThreshold, members, etcdClient, etcdClient, log)
+	if err != nil {
+		log.Error(err, "failed to defrag if necessary")
+	}
+
+}
+
+func (r *EtcdClusterReconciler) defragCronJob(log logr.Logger, cluster *etcdv1alpha1.EtcdCluster) {
+	ctx, cancel, etcdClient, members, ok := r.setupDefragDeps(log, cluster)
+	if !ok {
+		return
+	}
+	defer cancel()
+	defer etcdClient.Close()
+
+	ctx = context.Background()
```

Comment:
Here we overwrite the ctx we're meant to be using, which means we're not setting a timeout on our defrag call